### PR TITLE
Feature/8 oauth2

### PR DESCRIPTION
**Pull request by kandelk**
Add backend and frontend for oauth2. Keys are stored on trello. For accessing backend data client should pass token as Bearer

### CODE REVIEW

File: backend/core/src/main/java/com/knewless/core/auth/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
+
+import javax.validation.Valid;
+import java.net.URI;
+
+@RestController
+@RequestMapping("/auth")
+public class AuthController {
```
Comment (MaximGoncharukSW) [inline backend/core/src/main/java/com/knewless/core/auth/AuthController.java:20]: Can you move all the logic from this class into AuthService class?

File: backend/core/src/main/java/com/knewless/core/user/UserController.java [hunk 2]
```diff
@@ -1,10 +1,22 @@
...
+import com.knewless.core.exception.ResourceNotFoundException;
+import com.knewless.core.security.oauth.UserPrincipal;
+import com.knewless.core.user.model.CurrentUser;
+import com.knewless.core.user.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class UserController {
```
Comment (MaximGoncharukSW) [inline backend/core/src/main/java/com/knewless/core/user/UserController.java:11]: Can you move this logic from the controller into UserServices class?

File: frontend/src/screens/Authentication/constants.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export const API_BASE_URL = 'http://localhost:5000';
+export const ACCESS_TOKEN = 'accessToken';
+
+export const OAUTH2_REDIRECT_URI = 'http://localhost:3000/oauth2/redirect';
```
Comment (MaximGoncharukSW) [inline frontend/src/screens/Authentication/constants.tsx]: Make it relative:
`'/auth/redirect'`

File: frontend/src/screens/Authentication/constants.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export const API_BASE_URL = 'http://localhost:5000';
```
Comment (MaximGoncharukSW) [inline frontend/src/screens/Authentication/constants.tsx]: Get rid of the URL

File: frontend/src/screens/Authentication/constants.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+export const ACCESS_TOKEN = 'accessToken';
+
+export const OAUTH2_REDIRECT_URI = 'http://localhost:3000/oauth2/redirect';
+
+export const GOOGLE_AUTH_URL = `${API_BASE_URL}/oauth2/authorize/google?redirect_uri=${OAUTH2_REDIRECT_URI}`;
+export const FACEBOOK_AUTH_URL = `${API_BASE_URL}/oauth2/authorize/facebook?redirect_uri=${OAUTH2_REDIRECT_URI}`;
```
Comment (MaximGoncharukSW) [inline frontend/src/screens/Authentication/constants.tsx:9]: Remove prefixes

File: frontend/src/screens/Authentication/constants.tsx (hunk 4 above)
Comment (kandelk) [reply frontend/src/screens/Authentication/constants.tsx]: I am not sure I can remove this URL, cause it's used in link's href which leads to the backend authorization and href doesn't work with proxy

File: frontend/src/screens/Authentication/constants.tsx (hunk 5 above)
Comment (kandelk) [reply frontend/src/screens/Authentication/constants.tsx:9]: Here I also can't remove prefixes due to href and proxy

**Comment by kandelk** [comment]
The only thing I couldn't do is to pass env from the docker to the client, but local .env file works.

**Review by MaximGoncharukSW — APPROVED** [review]

**Pull request merged**